### PR TITLE
DuckDB SQL sources for async reconcile

### PR DESCRIPTION
**Pull request by begelundmuller**
- Moves all SQL source rewriting into the `duckdb_to_duckdb` transporter
- Keeping logic in old reconcile the same for now (verified this PR does not break the old logic)
- Removes embedded sources support from async reconcile

### CODE REVIEW

File: runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go [hunk 1]
```diff
@@ -33,6 +37,96 @@ func (t *duckDBToDuckDB) Transfer(ctx context.Context, srcProps, sinkProps map[s
...
+	// TODO: This is pretty hacky and we should ideally break the relevant object store functionality out into a util function that we can use here.
+	// (Or consider rethinking how object store connectors work in general.)
+	if uri.Scheme == "s3" || uri.Scheme == "gs" {
+		if uri.Scheme == "gs" {
+			uri.Scheme = "gcs"
+		}
+
+		conn, release, err := opts.AcquireConnector(uri.Scheme)
+		if err != nil {
+			return fmt.Errorf("sql references s3, but not able to acquire connector: %w", err)
```
Comment (begelundmuller) [inline runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go:78]:
```suggestion
			return fmt.Errorf("sql references %q, but not able to acquire connector: %w", uri.Scheme, err)
```

File: runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go [hunk 2]
```diff
@@ -33,6 +37,96 @@ func (t *duckDBToDuckDB) Transfer(ctx context.Context, srcProps, sinkProps map[s
 		return err
 	}
 
+	// We can't just pass the SQL statement to DuckDB outright.
+	// We need to do some rewriting for some table references (currently object stores and local files).
```
Comment (begelundmuller) [inline runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go:41]:
```suggestion
	// We need to do some rewriting for certain table references (currently object stores and local files).
```

File: runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go [hunk 3]
```diff
@@ -33,6 +37,96 @@ func (t *duckDBToDuckDB) Transfer(ctx context.Context, srcProps, sinkProps map[s
...
+
+		conn, release, err := opts.AcquireConnector(uri.Scheme)
+		if err != nil {
+			return fmt.Errorf("sql references s3, but not able to acquire connector: %w", err)
+		}
+		defer release()
+
+		objStore, ok := conn.AsObjectStore()
+		if !ok {
+			return fmt.Errorf("expected s3 connector to implement ObjectStore")
```
Comment (begelundmuller) [inline runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go:84]:
```suggestion
			return fmt.Errorf("expected connector %q to implement ObjectStore", uri.Scheme)
```

File: runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go [hunk 4]
```diff
@@ -33,6 +37,96 @@ func (t *duckDBToDuckDB) Transfer(ctx context.Context, srcProps, sinkProps map[s
...
+	// Parse the path as a URL (also works for local paths)
+	uri, err := url.Parse(ref.Paths[0])
+	if err != nil {
+		return fmt.Errorf("could not parse table function path %q: %w", ref.Paths[0], err)
+	}
+
+	// If the path is an object store reference, rewrite to objectStoreToDuckDB transporter.
+	// TODO: This is pretty hacky and we should ideally break the relevant object store functionality out into a util function that we can use here.
+	// (Or consider rethinking how object store connectors work in general.)
+	if uri.Scheme == "s3" || uri.Scheme == "gs" {
```
Comment (AdityaHegde) [inline runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go:71]: Add a TODO comment to add azure storage when it is added. @rakeshsharma14317

File: runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go [hunk 5]
```diff
@@ -33,6 +37,96 @@ func (t *duckDBToDuckDB) Transfer(ctx context.Context, srcProps, sinkProps map[s
...
+
+	// Parse AST
+	ast, err := duckdbsql.Parse(srcCfg.SQL)
+	if err != nil {
+		return fmt.Errorf("failed to parse sql: %w", err)
+	}
+
+	// Error if there isn't exactly one table reference
+	refs := ast.GetTableRefs()
+	if len(refs) != 1 {
```
[the remark this answers is not in the document]
Comment (begelundmuller) [reply runtime/drivers/duckdb/transporter/duckDB_to_duckDB.go:51]: Yeah, but since it's not a regression I think we should do it in a separate PR

**Pull request merged**